app/main.py

```diff
@@ -1,25 +1,79 @@
-from abc import ABC
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: int, owner: str) -> tuple:
+        return self.min_amount, self.max_amount
+
+    def __set__(self, instance: int, value: int) -> None:
+        raise AttributeError("Cannot set value directly")
+
+    def __set_name__(self, owner: str, name: str) -> None:
```

Comment:
Revise your type annotation, you could use `object` or `Any` to be non-specific for arguments but do not explicitly limit the type annotations to the wrong ones.